pkg/virt-handler/vm.go

```diff
@@ -2300,7 +2300,7 @@ func (d *VirtualMachineController) checkNetworkInterfacesForMigration(vmi *v1.Vi
 		return nil
 	}
 
-	if !netvmispec.IsPodNetworkWithMasqueradeBindingInterface(vmi.Spec.Networks, ifaces) {
+	if !d.clusterConfig.LiveMigrationBridgeBindingPodNetworkEnabled() && !netvmispec.IsPodNetworkWithMasqueradeBindingInterface(vmi.Spec.Networks, ifaces) {
```

Comment:
We might need a more fine granular approach here than a global feature gate. For example, with ovn-k not all VMs will be live migratable with bridge. Only those which use that /32 route. It will be technically possible for someone to use pod+bridge without a /32 which would not survive live migration.

Instead of a featuregate, what would you think about a per VMI tuning that allowed live migration with pod+bridge. Something like an annotation `kubevirt.io/allow-pod-bridge-network-live-migration` which would be set on a per VMI basis.

We'd use that annotation in hypershift when we're sure we want the VMs to be live migratable. 

Overtime, as we gain more confidence, we can transition that annotation to a value in the VMI spec. Starting with an annotation is the least risky approach though for now.